x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/cache/common/CacheFileTests.java

```diff
@@ -455,13 +452,41 @@ public void testCacheFileCreatedAsSparseFile() throws Exception {
                 greaterThanOrEqualTo(cacheFile.getLength())
             );
 
-            if (Constants.LINUX) {
+            if (blockSize > 0L) {
                 final long nbBlocks = (cacheFile.getLength() + blockSize - 1) / blockSize; // ceil(cacheFile.getLength() / blockSize)
-                assertThat(
-                    "Cache file size mismatches (block size: " + blockSize + ", number of blocks: " + nbBlocks + ')',
-                    sizeOnDisk.getAsLong(),
-                    equalTo(nbBlocks * blockSize)
-                );
+                final long expectedSize = nbBlocks * blockSize;
+                if (blockSize == fourKb) {
+                    assertThat(
+                        "Cache file size mismatches (block size: "
+                            + blockSize
+                            + ", number of blocks: "
+                            + nbBlocks
+                            + ", file length: "
+                            + cacheFile.getLength()
+                            + ')',
+                        sizeOnDisk.getAsLong(),
+                        equalTo(expectedSize)
+                    );
+                } else {
+                    // block size other than usual default block size of 4KB indicates that a special filesystem may be at use; in this
+                    // case we verify that the allocated size for the file stays within some upper limit of an extra block of 4096L bytes
+                    assertThat(
+                        "Cache file size mismatches with non default block size (block size: "
+                            + blockSize
+                            + ", number of blocks: "
+                            + nbBlocks
+                            + ", file length: "
+                            + cacheFile.getLength()
+                            + ')',
+                        sizeOnDisk.getAsLong(),
+                        allOf(greaterThanOrEqualTo(expectedSize), lessThanOrEqualTo(expectedSize + fourKb))
```

Comment:
I wonder if we're being too prescriptive here. We don't really care how big the file is on disk (we've already checked that it's no longer too small). It's kind of nice that you can often compute the size on disk by rounding up the actual size to the next multiple of `blockSize` but maybe we don't even need this assertion?

Reply:
Thinking about it this morning, I do agree. Being able to round up to the next `4KB` is just a consequence of generating files of 1MB max. I removed the whole assertion and just kept the `/mnt/secret` - with an additionnal comment.